src/ape/managers/chain.py

```diff
@@ -276,16 +282,21 @@ def poll_blocks(
             raise ValueError("'stop' argument must be in the future.")
 
         # Get number of last block with the necessary amount of confirmations.
-        latest_confirmed_block_number = self.height - required_confirmations
-        has_yielded_before_reorg = False
+        last_yielded_hash = None
+        last_yielded_number = None
+        block = None
 
         if start_block is not None:
-            # Front-load historically confirmed blocks.
-            yield from self.range(start_block, latest_confirmed_block_number + 1)
-            has_yielded_before_reorg = True
+            # Front-load historical blocks.
+            for block in self.range(start_block, self.height - required_confirmations + 1):
+                yield block
+
+        if block:
+            last_yielded_hash = block.hash
+            last_yielded_number = block.number
+            time_since_last = time.time()
 
```

Comment:
```suggestion
else:
    last_yielded_hash = None
    last_yielded_number = None

```

Might also need `time_since_last` here?